pkg/sql/sqltestutils/telemetry.go

```diff
@@ -212,37 +212,49 @@ func (tt *telemetryTest) RunTest(
 		// Report diagnostics once to reset the counters.
 		reportDiags(ctx)
 		_, err := db.Exec(td.Input)
-		var buf bytes.Buffer
+		var errorStr = ""
 		if err != nil {
-			fmt.Fprintf(&buf, "error: %v\n", err)
+			// This test can be used to check telemetry counters even when a statement
+			// fails (in which case the output contains the error as well).
+			errorStr = fmt.Sprintf("error: %v\n", err)
 		}
-		reportDiags(ctx)
-		last := tt.diagSrv.LastRequestData()
-		usage := last.FeatureUsage
-		keys := make([]string, 0, len(usage))
-		for k, v := range usage {
-			if v == 0 {
-				// Ignore zero values (shouldn't happen in practice)
-				continue
-			}
-			if !tt.allowlist.Match(k) {
-				// Feature key not in allowlist.
-				continue
+		// Internal background queries can set telemetry counters. Retry for a few
+		// times and stop when we get the expected result.
+		const retries = 10
+		expected := strings.TrimSpace(td.Expected)
```

Comment:
we have to detect this in a `-rewrite` to stop it retrying 10 times no?

Reply:
I don't see a big problem, we will choose the output of the last run.